### PR TITLE
MLDB-1759 fix err 500 when using * with GROUP BY

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: server/bound_queries.cc [hunk 1]
```diff
@@ -1449,6 +1449,13 @@ execute(RowProcessor processor,
     typedef std::map<RowKey, GroupMapValue> GroupByMapType;
     std::vector<GroupByMapType> accum(numBuckets);
 
+    for (const auto & c: select.clauses) {
+        if (typeid(*c.get()) == typeid(WildcardExpression)) {
```
Comment (jeremybarnes) [inline server/bound_queries.cc]: I would implement an isWildcardExpression() function in SqlExpression, with a default implementation that returns false, and override this for WildcardExpression.  It reduces coupling and will allow (for example) a derivative of WildcardExpression to work properly.  Direct typeid checks like that should usually be replaced with a virtual method.

**Comment by jeremybarnes** [comment]
Right direction, but will be better with a virtual method.

File: server/bound_queries.cc (hunk 1 above)
Comment (jeremybarnes) [inline server/bound_queries.cc]: Also, a test based upon dynamic_cast is also more reliable as it will work for derived classes.

Comment (FinchPowers) [inline server/bound_queries.cc]: Bon ben fascinant. Ladite fonction existait déjà. :p

File: server/bound_queries.cc [hunk 2]
```diff
@@ -1449,6 +1449,13 @@ execute(RowProcessor processor,
     typedef std::map<RowKey, GroupMapValue> GroupByMapType;
     std::vector<GroupByMapType> accum(numBuckets);
 
+    for (const auto & c: select.clauses) {
+        if (c->isWildcard()) {
```
Comment (jeremybarnes) [inline server/bound_queries.cc]: You shouldn't use a function like isWildcard().  It makes the API too heavy.  You should compare the typeid() instead.

Comment (FinchPowers) [inline server/bound_queries.cc]: Are you trolling me?

Comment (jeremybarnes) [inline server/bound_queries.cc]: Yes, I was trolling you.  +1 to merge.

Comment (FinchPowers) [inline server/bound_queries.cc]: Haha. Je ne suis pas assez habitué à ça ;)

**Pull request merged**